website/docs/r/project_api_key.html.markdown

```diff
@@ -57,24 +57,19 @@ resource "mongodbatlas_project_api_key" "test" {
 
 ~> **NOTE:** Project created by API Keys must belong to an existing organization.
 
-# project_assignment
+### project_assignment
 Project Assignment attribute is optional (Use project_assignment going forward as role_names parameter above is deprecated)
 
 * `project_id` - (Required) Project ID to assign to Access Key
-* `role_names` - Name of the role. This resource returns all the roles the user has in Atlas.
+* `role_names` - (Required) List of Project roles that the Programmatic API key needs to have. Ensure you provide: at least one role and ensure all roles are valid for the Project.  You must specify an array even if you are only associating a single role with the Programmatic API key.
 The following are valid roles:
   * `GROUP_OWNER`
   * `GROUP_READ_ONLY`
   * `GROUP_DATA_ACCESS_ADMIN`
   * `GROUP_DATA_ACCESS_READ_WRITE`
   * `GROUP_DATA_ACCESS_READ_ONLY`
-  * `GROUP_CLUSTER_MANAGER`  
-  * 
-### Programmatic API Keys
-api_keys allows one to assign an existing organization programmatic API key to a Project. The api_keys attribute is optional.
+  * `GROUP_CLUSTER_MANAGER`
```

Comment:
can we remove this list of role and add a link to the documentation? https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/#tag/Programmatic-API-Keys/operation/addProjectApiKey

Reply:
same in other part of the file and in the data source documentation

Reply:
good point, adjusted. Used the following link https://www.mongodb.com/docs/atlas/reference/user-roles/#project-roles for referencing